src/globus_sdk/exc/warnings.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+import os
+import typing as t
+import warnings
+
+
+# this is (by meaning) a type of DeprecationWarning, but because python suppresses
+# DeprecationWarning by default, most users would never see it if we inherited from that
```

Comment:
> by default, most users would never see it

To be clear, that is the point. _Developers_ are the ones who should be seeing these warnings and addressing them as they will. By inheriting from `Warning` we are guaranteeing that those developers' _users_ have to see these warnings. I disagree with this outcome.

I think this should be a `DeprecationWarning` so we're not forcing warning messages upon our developers' users.

Reply:
This is similar to the argument which has underpinned the decision to ignore `DeprecationWarning` by default in the interpreter.

While I respect this position, I have a significant concern about this is how it impacts users who are writing relatively small scripts and have no testing mechanisms. Such a user will typically _never_ run the interpreter with `-W` filters specified.

`Warning` as the base + the toggle env var ensures that these users will see the warnings when we flip the default for the env var flag.
The big alternative I see is to use `DeprecationWarning` as the base and remove the env var (why bother suppressing the warning explicitly if it's a `DeprecationWarning`?).

i.e. I think that for our user-base, the developer/user distinction is less clear than it is with some other libraries. We can and should discuss this further. I could be convinced, in part because there's a question of how often and why/when scripting users would be updating to the newest library version. And the end-goal is to have the smoothest upgrade experience possible.

Reply:
One of the key points which got me onboard with using `DeprecationWarning` as the base, and getting the standard Python behavior, is that doing so provides the best experience for anyone who is doing their work and using pytest to test their code.